source/Plugins.cpp

```diff
@@ -345,6 +507,132 @@ const Set<Plugin> &Plugins::Get()
 // Toggles enabling or disabling a plugin for the next game restart.
 void Plugins::TogglePlugin(const string &name)
 {
+	lock_guard<mutex> guard(pluginsMutex);
 	auto *plugin = plugins.Get(name);
 	plugin->currentState = !plugin->currentState;
 }
+
+
+
+future<void> Plugins::Install(InstallData *installData, bool update)
+{
+	oldNetworkActivity = true;
+	if(!update)
+	{
+		lock_guard<mutex> guard(activePluginsMutex);
+		if(!activePlugins.insert(installData->name).second)
+			return future<void>();
+	}
+
+	return async(launch::async, [installData, update]() noexcept -> void
+		{
+			string zipLocation = Files::Plugins() + installData->name + ".zip";
```

Comment:
Same thing about using the untrusted name.

Reply:
I added a check here for a "..", I dont know what else I should test